crates/mysten-network/src/multiaddr.rs

```diff
@@ -112,3 +127,22 @@ pub(crate) fn parse_unix(address: &Multiaddr) -> Result<(Cow<'_, str>, &'static
 
     Ok((path, http_or_https))
 }
+
+#[cfg(test)]
+mod test {
+    use super::to_socket_addr;
+    use multiaddr::multiaddr;
+
+    #[test]
+    fn test_to_socket_addr() {
```

Comment:
Would it be worth adding a negative case for e.g. the `Dns` variant of `Protocol`? 

I want to help direct folks to the nearby `client::endpoint_from_multiaddr`, which does a bunch of similar things, and this (name resolution) is where the present function and that one differ.

Reply:
Sounds good. Added a test and a comment for `client::endpoint_from_multiaddr()`